src/main/java/Duke.java

```diff
@@ -1,10 +1,235 @@
+     */
+    public Duke(String filePath) {
+        try {
+            storage = new Storage(filePath);
+            tasks = new TaskList(storage.loadFile()); // add existing tasks from txt file onto ArrayList
+            size = tasks.getSize(); // get existing size of list
+            storage.closeFile();
+        } catch (DukeException e) {
+            ui.loadError();
+            ui.createFile();
+            ArrayList<Task> list = new ArrayList<>();
+            tasks = new TaskList(list);
+            File file = new File(filePath); // create new .txt file with provided filepath
+            try {
+                file.createNewFile();
+            } catch (Exception err) {
+                System.out.println("Unable to create new file."); // Unlikely
+            }
+        }
+    }
+
+    /**
+     * Run the Duke application.
+     * Accepted commands: list, done, delete, find
+     * Accepted tasks: todo, event, deadline
+     */
+    public void run() {
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        // Store command-line input as String
+        String input = "";
+
+        Storage storage = this.storage;
+        TaskList tasks = this.tasks;
+        Parser parser = new Parser();
+
+        // Create new user interface object
+        Ui ui = new Ui();
+
+        // Read command-line input with Scanner
+        Scanner scanner = new Scanner(System.in);
+
+        // Initial opening introduction and prompt for user input
+        ui.openingStatement();
+
+        // Check for last statement
+        while (!input.equals("bye")) {
+
+            // Get entire line of input from command-line
+            input = scanner.nextLine().trim(); //Remove blank space
+
+            // Store whatever text entered, except "bye", exit loop
+            if (input.equals("bye")) {
+                break;
+            }
+            ui.separator();
+
+            if (input.equals("list")) {
+                ui.showList();
+                tasks.printList();
+            } else if (input.contains("done")) {
+
+                try {
+                    //Mark task as done
+                    tasks.setDone(input);
+                } catch (NullPointerException err) {
+                    ui.invalidEntry();
+                    continue;
+                }
+
+            } else if (input.contains("delete")) {
+
+                try {
+                    //Mark task as done
+                    tasks.deleteTask(input);
+                } catch (NullPointerException err) {
+                    ui.invalidEntry();
+                    continue;
+                }
+            } else if (input.contains("todo") || input.contains("deadline") || input.contains("event")) {
```

Comment:
Hmmm if the description of a task contains any of these keywords, wouldn't the task fall into the wrong category?
eg. deadline complete todo list /by 12/12/2020 0800; here the task will fulfill the `input.contains("todo")` condition and go through the todo handler.

Reply:
Ah yes, thanks for the good observation! Changed it to checking the first string for the command instead!